File: 25/12/go/3075.go
package A

import (
	"slices"
)

func maximumHappinessSum(happiness []int, k int) int64 {
	ans := int64(0)
	slices.SortFunc(happiness, func(a, b int) int {
		return b - a
	})
	for i, v := range happiness {
		if v < i || i > k-1 {
			break
		}
		ans += int64(v - i)
	}
	return ans
}

File: 25/12/3075.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
  long long maximumHappinessSum(std::vector<int> &happiness, int k) {
    long long ans = 0;
    std::sort(happiness.begin(), happiness.end(),
              [](int a, int b) { return a > b; });
    for (int i = 0; i < k && i < happiness.size(); i++) {
      if (happiness[i] < i) {
        break;
      }
      ans += happiness[i] - i;
    }
    return ans;
  }
};